Lib/test/test_pkgutil.py

```diff
@@ -92,6 +93,44 @@ def test_getdata_zipfile(self):
 
         del sys.modules[pkg]
 
+    def test_issue44061_iter_modules(self):
+        #see: issue44061
+        zip = 'test_getdata_zipfile.zip'
+        pkg = 'test_getdata_zipfile'
+
+        # Include a LF and a CRLF, to test that binary data is read back
+        RESOURCE_DATA = b'Hello, world!\nSecond line\r\nThird line'
+
+        # Make a package with some resources
+        zip_file = os.path.join(self.dirname, zip)
+        z = zipfile.ZipFile(zip_file, 'w')
+
+        # Empty init.py
+        z.writestr(pkg + '/__init__.py', "")
+        # Resource files, res.txt
+        z.writestr(pkg + '/res.txt', RESOURCE_DATA)
+        z.close()
+
+        # Check we can read the resources
+        sys.path.insert(0, zip_file)
+        res = pkgutil.get_data(pkg, 'res.txt')
+        self.assertEqual(res, RESOURCE_DATA)
+
+        # make sure iter_modules accepts Path objects
+        names = []
+        for moduleinfo in pkgutil.iter_modules([Path(zip_file)]):
+            self.assertIsInstance(moduleinfo, pkgutil.ModuleInfo)
+            names.append(moduleinfo.name)
+        self.assertEqual(names, [pkg])
+
+        del sys.path[0]
+        del sys.modules[pkg]
```

Comment:
Cleanup should at least happen inside a `finally` block that includes everything from when it was changed. There might be better helpers for this, but I don't recall off the top of my head.

Otherwise, if the test fails, it could break other tests as well.

Reply:
I see, agree, it makes sense!

I actually based this test on a very similar and having a look at this `Lib/test/test_pkgutil.py` most of the cleanups look the same.

I pushed a change, can you please have a look again?